core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/solver/event/AbstractEventSupport.java

```diff
@@ -1,19 +1,45 @@
 package org.optaplanner.core.impl.solver.event;
 
+import java.util.Collection;
 import java.util.EventListener;
-import java.util.Set;
-import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.concurrent.atomic.AtomicLong;
 
 public class AbstractEventSupport<E extends EventListener> {
 
-    protected Set<E> eventListenerSet = new CopyOnWriteArraySet<>();
+    private final AtomicLong ID_COUNTER = new AtomicLong();
```

Comment:
Brevity is not always clarity. My mistake. This is a code style issue in my opinion. Even if it was `static` it still wouldn't be a constant because the counter is mutable and it _changes_. So, it's not a constant and should not use the `UPPER_CASE` style.